src/repl.rs

```diff
@@ -58,12 +58,79 @@ pub trait ReplTrait<F: LurkField> {
 
     fn command() -> Command;
 
+    fn handle_form<P: AsRef<Path> + Copy, T: Iterator<Item = char>>(
+        &mut self,
+        store: &mut Store<F>,
+        chars: &mut peekmore::PeekMoreIterator<T>,
+        package: &Package,
+        pwd: P,
+        update_env: bool,
+    ) -> Result<()> {
+        let (ptr, is_meta) = store.read_maybe_meta(chars, package)?;
+
+        if is_meta {
+            let pwd: &Path = pwd.as_ref();
+            self.handle_meta(store, ptr, package, pwd)
+        } else {
+            self.handle_non_meta(store, ptr, update_env).map(|_| ())
+        }
+    }
+
     fn handle_run<P: AsRef<Path> + Copy>(
         &mut self,
         store: &mut Store<F>,
         file_path: P,
         package: &Package,
-    ) -> Result<()>;
+    ) -> Result<()> {
+        eprintln!("Running from {}.", file_path.as_ref().to_str().unwrap());
+        self.handle_file(store, file_path, package, false)
+    }
+
+    fn handle_load<P: AsRef<Path>>(
+        &mut self,
+        store: &mut Store<F>,
+        file_path: P,
+        package: &Package,
+    ) -> Result<()> {
+        eprintln!("Loading from {}.", file_path.as_ref().to_str().unwrap());
+        self.handle_file(store, file_path.as_ref(), package, true)
+    }
+
+    fn handle_file<P: AsRef<Path> + Copy>(
+        &mut self,
+        store: &mut Store<F>,
+        file_path: P,
+        package: &Package,
+        update_env: bool,
+    ) -> Result<()> {
+        let file_path = file_path;
+
+        let input = read_to_string(file_path)?;
+        eprintln!(
+            "Read from {}: {}",
+            file_path.as_ref().to_str().unwrap(),
+            input
+        );
+        let mut chars = input.chars().peekmore();
+
+        loop {
+            if let Err(e) = self.handle_form(
+                store,
+                &mut chars,
+                package,
+                // use this file's dir as pwd for further loading
+                file_path.as_ref().parent().unwrap(),
```

Comment:
The examples on parent make me think some innocuous paths may blow this unwrap up:
https://doc.rust-lang.org/std/path/struct.Path.html#examples-10

Reply:
That could be, but this is unchanged in the current work, so let's not worry about it now.